lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -387,6 +388,17 @@ def configure(conf)
                   ' ignored.'
       end
 
+      if @enable_metadata_agent
+        # 1. If @metadata_agent_url is customized, use that.
+        # 2. Otherwise check the environment variable
+        #    STACKDRIVER_METADATA_AGENT_URL.
+        # 3. Fall back to the default
+        if @metadata_agent_url == DEFAULT_METADATA_AGENT_URL && \
+           !ENV[METADATA_AGENT_URL_ENV_VAR_NAME].to_s.empty?
```

Comment:
Convert the env variable to string to capture empty strings (in which case it should be considered not set).